openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py

```diff
@@ -2,186 +2,162 @@
+    }
 
-HEADERS = {"api_key": API_KEY}
+    def __init__(self, conf: dict = None, date: str = None):
+        super().__init__(conf, date)
+        self.api_key = Variable.get("API_KEY_BROOKLYN_MUSEUM")
+        self.headers = {"api_key": self.api_key}
 
-DEFAULT_QUERY_PARAMS = {
-    "has_images": 1,
-    "rights_type_permissive": 1,
-    "limit": LIMIT,
-    "offset": 0,
-}
+    def get_media_type(self, record: dict) -> str:
+        return constants.IMAGE
 
-
-def main():
-    logger.info("Begin: Brooklyn museum provider script")
-    offset = 0
-    condition = True
-    while condition:
-        query_param = _get_query_param(offset=offset)
-        objects_batch = _get_object_json(query_param=query_param)
-        logger.debug(f"Number of batches: {len(objects_batch or [])}")
-        if type(objects_batch) == list and len(objects_batch) > 0:
-            _process_objects_batch(objects_batch)
-            logger.debug(f"Images till now {image_store.total_items}")
-            offset += LIMIT
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        if not prev_query_params:
+            offset = 0
         else:
-            condition = False
-    image_store.commit()
-    logger.info(f"Total images received {image_store.total_items}")
-
-
-def _get_query_param(offset=0, default_query_param=None):
-    if default_query_param is None:
-        default_query_param = DEFAULT_QUERY_PARAMS
-    query_param = default_query_param.copy()
-    query_param.update(offset=offset)
-    return query_param
-
-
-def _get_object_json(
-    headers=None, endpoint=ENDPOINT, retries=RETRIES, query_param=None
-):
-    if headers is None:
-        headers = HEADERS.copy()
-    data = None
-    for tries in range(retries):
-        response = delay_request.get(endpoint, query_param, headers=headers)
-        try:
-            response_json = response.json()
-            if response_json and response_json.get("message", "").lower() == "success.":
-                data = response_json.get("data")
-                break
-        except (
-            RequestException,
-            AttributeError,
-            JSONDecodeError,
-            ValueError,
-            TypeError,
-        ) as e:
-            logger.error(f"Error due to {e}")
-    return data
-
-
-def _process_objects_batch(objects_batch):
-    for object_ in objects_batch:
-        rights_info = object_.get("rights_type")
-        license_url = _get_license_url(rights_info)
+            offset = kwargs.get("offset", 0) + self.batch_limit
+        return {**self._common_args, "offset": offset}
+
+    @staticmethod
+    def _get_data_from_response(response_json) -> dict | None:
+        if response_json and response_json.get("message", "").lower() == "success.":
+            return response_json.get("data")
+        return None
+
+    def get_should_continue(self, response_json):
+        # We can assume that any truthy response from the get data function would
```

Comment:
This is the kind of thing where I would want to put a TO DO, with a reference to https://github.com/WordPress/openverse-catalog/issues/385

Reply:
Very smart!